src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6062,6 +6062,11 @@ private bool IsUsingAliasInScope(string name)
                         }
                     default:
                         {
+                            // Ensure that there is an error thrown when an Unbound Generic is found.
+                            if (right is GenericNameSyntax genericNameRight && rightHasTypeArguments && typeArgumentsSyntax.Any(SyntaxKind.OmittedTypeArgument) && !IsUnboundTypeAllowed(genericNameRight))
```

Comment:
Do we need to do this for _every_ `default` case, or just the cases that return from here?

Reply:
You're right, only the ones that return from here need to be affected. Moved to a more specific scope.